Documentation/dev-guide/api_grpc_gateway.md

```diff
@@ -0,0 +1,38 @@
+
+## Why grpc-gateway
+
+etcd v3 uses [gRPC][grpc]. Most recommended is to use etcd [Go client][go-client] and [etcdctl][etcdctl]. However, some languages are not supported by gRPC. This is where [grpc-gateway][grpc-gateway] steps in. It builds a RESTful proxy based on the gRPC services, which can serve HTTP and JSON requests.
+
+
+## How grpc-gateway
+
+gRPC messages are in Protocol Buffer, serialized binary data. So requests to grpc-gateway need to be base64-encoded.
```

Comment:
The gateway accepts the [JSON mapping](https://developers.google.com/protocol-buffers/docs/proto3#json) for the messages defined in [Protocol Buffer](link to our protobuf api doc) format.